test/helper.js

```diff
@@ -0,0 +1,32 @@
+var async = require('async'),
+    net = require('net');
+
+
+function createServer(base, host, next) {
+  var server = net.createServer(function () {});
+
+  if (!next) {
+    server.listen(base, host);
+  } else {
+    server.listen(base, host, next);
+  }
+
+  server.on('error', function() {
+    server.close();
+  });
```

Comment:
@indexzero - 

```
server.on('error', function() {
  server.close();
});
```

does not work in node v0.10 - neither does putting that inside on('listening') - otherwise this passes now (on 0.12 and node 4.x)

any thoughts?